test/functional/pdo_sqlsrv/pdo_ae_output_param_char_size.phpt

```diff
@@ -0,0 +1,173 @@
+"22003" => "Numeric value out of range");
+ 
+$pdoParamTypes = array(
+    PDO::PARAM_BOOL,    // 5
+    PDO::PARAM_NULL,    // 0
+    PDO::PARAM_INT,     // 1 
+    PDO::PARAM_STR,     // 2
+    PDO::PARAM_LOB      // 3
+);
+
+//////////////////////////////////////////////////////////////////////////////////
+function printValues($msg, $det, $rand, $input0, $input1)
+{
+    echo $msg;
+    echo "input 0: "; var_dump($input0); 
+    echo "fetched: "; var_dump($det); 
+    echo "input 1: "; var_dump($input1);
+    echo "fetched: "; var_dump($rand); 
+}
+
+function testOutputChars($inout)
+{
+    global $pdoParamTypes, $dataTypes, $lengths, $errors;
+    
+    try {
+        
+        $conn = connect();
+        $tbname = "test_char_types";
+        $spname = "test_char_proc";
+        
+        foreach ($dataTypes as $dataType) {
+            $maxtype = strpos($dataType, "(max)");
+            foreach ($lengths as $length) {
+                if ($maxtype !== false) {
+                    $type = $dataType;
+                } else {
+                    $type = "$dataType($length)";
+                }
+                trace("\nTesting $type:\n");
+                    
+                //create and populate table
+                $colMetaArr = array(new ColumnMeta($type, "c_det"), new ColumnMeta($type, "c_rand", null, "randomized"));
+                createTable($conn, $tbname, $colMetaArr);
+                $input0 = str_repeat("1", $length);
+                $input1 = str_repeat("2", $length);
+                insertRow($conn, $tbname, array("c_det" => $input0,
+                                                "c_rand" => $input1));
+                    
+                // fetch with PDO::bindParam using a stored procedure
+                dropProc($conn, $spname);
+                $spSql = "CREATE PROCEDURE $spname (
+                                @c_det $type OUTPUT, @c_rand $type OUTPUT ) AS
+                                SELECT @c_det = c_det, @c_rand = c_rand FROM $tbname";
+                $conn->query($spSql);
+                
+                // call stored procedure
+                $outSql = getCallProcSqlPlaceholders($spname, 2);
+                foreach ($pdoParamTypes as $pdoParamType) {
+                    $det = "";
+                    $rand = "";
+                    $stmt = $conn->prepare($outSql);
+                    trace("\nParam $pdoParamType with INOUT = $inout\n");
+                    
+                    if ($inout) {
+                        $paramType = $pdoParamType | PDO::PARAM_INPUT_OUTPUT;
+                    } else {
+                        $paramType = $pdoParamType;
+                    }
+                    
+                    $len = $length;
+                    if ($pdoParamType == PDO::PARAM_BOOL || $pdoParamType == PDO::PARAM_INT) {
+                        $len = PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE;
+                        $det = $rand = 0;
+                    }

```

Comment:
so based on the PDO::PARAM_ type, you initialize the output variable differently. Then maybe you should also initialize $det = $rand = NULL for PDO::PARAM_NULL

Reply:
But binding output as PARAM NULL always throws exception